### DISCUSSION TITLE
redirect ```eth_sendTransaction``` RPC calls

### DISCUSSION THREAD

**Discussion by Arch0125** [MetaMask/snaps#1272] [Q&A]
Is there any possible to make MetaMask ask whether to sign the Tx through SCW or EOA, i.e. create a choice popup before normal Tx ui pops because of onTransactionHandler

**Comment by ziad-saab** [comment]
This is not currently possible. You have to implement your transaction flow in a completely custom way inside a Dapp.

**Comment by Arch0125** [comment]
Also another question, is there a way to reject a transaction automatically inside onTransactionHandler block ?

**Reply by Arch0125** [reply]
A bit of context about what im trying to achieve 
-> Transaction popup appears on ```eth_sendTransaction```
-> On clicking the Transaction insights tab same tx window pops open, when rejected it opens the custom scw signing screen written inside the onTransactionHandler block

https://user-images.githubusercontent.com/72722967/224569370-76ab8a8d-a312-4714-85d6-01196e151a34.mov

**Comment by ziad-saab** [comment]
You cannot control the flow of a transaction using MetaMask Snaps. The only thing you can do is _display_ insights when the user switches to your tab in the confirmation window. Your `onTransaction` hook will not even see the transaction if the user doesn't switch to your tab.

**Reply by Arch0125** [reply]
Any thoughts regarding this UX ?

https://user-images.githubusercontent.com/72722967/224571864-8761b85b-d006-4730-b435-9b55d14c7cc3.mov

**Reply by ziad-saab** [reply]
I haven't looked at your code, but logically this will only work when code from a Dapp you control initiates the transaction flow. You wouldn't be able to detect a rejected transaction that you haven't initiated yourself. Since that's the case, your Dapp should be able to shortcut this whole flow and simply ask the user to sign a SCW transaction directly. Am I missing something?

**Reply by LaplaceMan** [reply]
Hello, I'm working on a similar Snap for SCW and I'm more curious about how your signature is implemented, Snap doesn't seem to be able to use the private key of the EOA in Metamask.

**Reply by Montoya** [reply]
That's correct, the Snap cannot use the private key of an EOA. 

If you are interested in building a SCW Snap, keep an eye out for a Keyrings API that will be landing in our docs by next week.
